tests/contracts/replicate.scilla

```diff
@@ -0,0 +1,10 @@
+scilla_version 0
+
+contract Foo ()
+
+transition rep (bar : ByStr20 with contract end)
+  foo = { _replicate_contract : "" };
```

Comment:
I wouldn't mind if we had some tests with init parameters (that differ from the ones in the existing contract), and at least two failing tests: One where the contract to be replicated doesn't exist, and one where the replicate parameters don't match the contract parameters of the replicated contract.

We should also test that it works to send a message to a newly replicated contract, i.e.,
```
new_contract <-& REPLICATE_CONTRACT (old_addr, input);
msg = { _tag : "TransitionName"; _recipient : new_contract ; ... };
send msg;
```
The reason I would like this to be tested is that the replicated contract only exists in the temporary state of the blockchain, and we need to make sure that we can actually interact with contracts in the temporary state. This test obviously needs to be incorporated into the blockchain testsuite, because we can't test it only on the OCaml side.